tests/MongoDB.Driver.Tests/JsonDrivenTests/JsonDrivenTestFactory.cs

```diff
@@ -75,6 +98,20 @@ public JsonDrivenTest CreateTest(string receiver, string name)
                         case "assertSessionUnpinned": return new JsonDrivenAssertSessionUnpinnedTest(_testRunner, _objectMap);
                         case "assertSameLsidOnLastTwoCommands": return new JsonDrivenAssertSameLsidOnLastTwoCommandsTest(_testRunner, _eventCapturer, _objectMap);
                         case "assertSessionTransactionState": return new JsonDrivenAssertSessionTransactionStateTest(_testRunner, _objectMap);
+                        case "wait": return new JsonDrivenWait(_testRunner, _objectMap);
+                        case "waitForEvent": return new JsonDrivenWaitForEvent(_testRunner, _objectMap, _eventCapturer);
+                        case "assertEventCount": return new JsonDrivenAssertEventsCount(_testRunner, _objectMap, _eventCapturer);
+                        case "startThread": return new JsonDrivenStartThread(_testsContext, _testRunner, _objectMap);
+                        case "runAdminCommand":
+                            {
+                                var commandSettings = _client.Settings.Clone();
```

Comment:
the spec requires to run this command on the different client https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/tests/README.rst#runadmincommand.

Reply:
I would rather see `JsonDrivenRunAdminCommand` create the client itself and immediately Dispose of it itself after using it. Then `JsonDrivenTest` would not have to be `IDisposable`.

Reply:
`ccommandSettings`?

These are `clientSettings`.

Reply:
>clientSettingsForRunCommand 

done. But i will try to find a better solution here

Reply:
I've changed this code. To be honest I like the previous solution more than the current since it allows reusing the code which actually makes the same thing(calling `runCommand`) and it worked more similar to the previous jsontest implementations where the only logic which we do in the `Act` methods is calling the tested method itself without any additional logic like test preparation or so. I also don't see any problems if JSON test step has a Dispose method.
But in interests of time, I applied the suggestion.

I will rely on @rstam  and @vincentkam opinion here.